src/Fabulous/StackAllocatedCollections.fs

```diff
@@ -1,132 +1,581 @@
 module Fabulous.StackAllocatedCollections
 
-module Array =
-    let inline appendOne (v: 'v) (arr: 'v array) =
-        let res = Array.zeroCreate(arr.Length + 1)
-        Array.blit arr 0 res 0 arr.Length
-        res.[arr.Length] <- v
-        res
+open System
+open System.Collections.Generic
+open System.Runtime.CompilerServices
+
 
 
 //let a = Array.length [||]
 //[<System.Runtime.CompilerServices.IsReadOnly>]
-type StackArray3<'v> = (struct (int * 'v * 'v * 'v * 'v array))
-
-//    struct
-//        val count: int
-//        val v0: 'v
-//        val v1: 'v
-//        val v2: 'v
-//        val rest: 'v array
+
+
+module StackList =
+    type private Items<'v> = (struct ('v * 'v * 'v))
+
+    module private Items =
+        let inline one<'v> (v: 'v) =
+            Items(v, Unchecked.defaultof<'v>, Unchecked.defaultof<'v>)
+
+        let inline two<'v> (v1: 'v, v2: 'v) = Items(v1, v2, Unchecked.defaultof<'v>)
+
+    /// reference type that holds the chain of tuples that come before Data
+    /// this is basically List
+    [<NoComparison; NoEquality>]
+    type private Part<'v> =
+        | Empty
+        | Filled of struct (Items<'v> * Part<'v>)
+
+    module private Part =
+        let inline combine (before: Part<'v>, after: Part<'v>) : Part<'v> = Empty
```

Comment:
What's the usage of this function? It always returns Empty

Reply:
Nop, leftover from other ideas I had. Should delete it